### DISCUSSION TITLE
Windows clipboard history

### DISCUSSION THREAD

**Discussion by esndb** [AndrewRathbun/DFIRArtifactMuseum#4] [Ideas]
Please, add the Windows clipboard history data located at <USERPROFILE>\AppData\Local\Microsoft\Windows\Clipboard. Both Pinned and HistoryData folders.

**Comment by AndrewRathbun** [comment]
I thought Windows Clipboard data was stored within the ActivitiesCache.db SQLite database. I'm not seeing the Clipboard folder (or anything beyond) on my everyday Windows 10 machine.

**Reply by AndrewRathbun** [reply]
![image](https://user-images.githubusercontent.com/36825567/157872785-1b44af34-5b49-48b1-a6f0-c9bb20b1918f.png)

I'm guessing this needs to be enabled?

**Reply by AndrewRathbun** [reply]
![image](https://user-images.githubusercontent.com/36825567/157873160-23033039-7aaf-4bc7-b803-68c90ba6b4bb.png)

Got it enabled now. I'll see what I can do to generate some data, thank you for the idea!

**Reply by AndrewRathbun** [reply]
@esndb try to respond in this thread so we can keep things organized :)

do you know how to populate the HistoryData folder? I can only get data populated in the Pinned folder currently.

**Reply by AndrewRathbun** [reply]
@esndb https://github.com/AndrewRathbun/DFIRArtifactMuseum/tree/main/Windows/WindowsClipboard/Windows10/RathbunVM/Clipboard/Pinned/%7B4CA90608-9262-43D0-9045-FBB7C43B026F%7D is what I got so far

**Reply by esndb** [reply]
So did I. No idea what the HistoryData is for. Trying to figure it out, but no progress so far.

**Reply by AndrewRathbun** [reply]
Are you signed into a MS account? I am not in my VM. I wonder if that makes a difference. 

https://thinkdfir.com/2018/10/14/clippy-history/ should have some more insight.

**Reply by AndrewRathbun** [reply]
CC: @randomaccess3

**Reply by esndb** [reply]
Yeah, a sort of dpapi encryption. My MS account does not populate anything other than the "pinned" folder.

**Reply by randomaccess3** [reply]
The clipboard data is in memory unless you pin stuff I think. 
Maybe on sync it's stored on disk as well, can't remember, need to reread the blog post

I couldn't get the DPAPI (new version) to decrypt it though.
